app/utils/webhooks.py

```diff
@@ -69,10 +127,166 @@ async def _get_sticker_embed(sticker: discord.StickerItem) -> discord.Embed:
+        assert not isinstance(
+            forward.channel, discord.DMChannel | discord.PartialMessageable
+        )
+        embed.set_footer(text=f"#{forward.channel.name}")
+
+    images = [
+        attachment
+        for attachment in msg_data.attachments
+        if Path(attachment.filename).suffix in SUPPORTED_IMAGE_FORMATS
+    ]
+    image_only_embeds = [
+        embed
+        for embed in embeds
+        if not embed.title
+        if not embed.description
+        if not embed.fields
+        if embed.image
+    ]
+    if len(images) == 1 or len(image_only_embeds) == 1:
+        if images:
+            # https://discordpy.readthedocs.io/en/stable/faq.html#how-do-i-use-a-local-image-file-for-an-embed-image
+            embed.set_image(url="attachment://" + images[0].filename)
+        else:
+            image = image_only_embeds[0].image
+            # Try both, as embeds which have not yet been sent to Discord will
+            # not have a proxy_url.
+            embed.set_image(url=image.proxy_url or image.url)
+            embeds.remove(image_only_embeds[0])
+    if embeds or len(msg_data.attachments) > (1 if images else 0):
+        embed.add_field(
+            name="", value="-# (other forwarded content is attached)", inline=False
+        )
+
+    for line in _format_subtext(
+        None,
+        msg_data,
+        include_timestamp=False,
+    ).splitlines()[1:]:
+        embed.add_field(name="", value=line, inline=False)
+    embed.add_field(name="", value=f"-# [**Jump**](<{link}>) 📎", inline=False)
+
+    embeds.insert(0, embed)
+    return embeds, msg_data.attachments
+
+
+def _format_missing_reference(
+    message: discord.Message,
+) -> discord.Embed:
+    assert message.reference is not None
+    if message.reference.type == discord.MessageReferenceType.forward:
+        return discord.Embed(description="*Forwarded message was deleted.*").set_author(
+            name="➜ Forwarded"
+        )
+    return discord.Embed(description="*Original message was deleted.*").set_author(
+        name=(
+            "⚡️ Message"
+            if message.type == discord.MessageType.context_menu_command
+            else "↪️ Reply"
+        )
+    )
+
+
+def _format_interaction(message: discord.Message) -> str:
+    if not message.interaction_metadata:
+        return message.content
+    # HACK: Message.interaction is deprecated, and discord.py disables any
+    # warning filter resulting in a bunch of warnings spammed in the logs even
+    # if it is ignored. There is no other way to get the name, and
+    # Message._interaction is not marked deprecated. Delectable.
+    if hasattr(message, "_interaction") and message._interaction is not None:  # pyright: ignore [reportPrivateUsage] # noqa: SLF001
+        prefix = "/" * (message.type != discord.MessageType.context_menu_command)
+        name = f"`{prefix}{message._interaction.name}`"  # pyright: ignore [reportPrivateUsage] # noqa: SLF001
```

Comment:
I'm pretty sure we don't need the `hasattr()` call?
```suggestion
    if message._interaction is not None:  # pyright: ignore [reportPrivateUsage] # noqa: SLF001
        prefix = "/" * (message.type != discord.MessageType.context_menu_command)
        name = f"`{prefix}{message._interaction.name}`"  # pyright: ignore [reportPrivateUsage] # noqa: SLF001
```
Also, since `MessageInteraction`s seem to always be truthy, I think we can bind it to avoid another warning suppress 2 lines later:
```suggestion
    if interaction := message._interaction:  # pyright: ignore [reportPrivateUsage] # noqa: SLF001
        prefix = "/" * (message.type != discord.MessageType.context_menu_command)
        name = f"`{prefix}{interaction.name}`"
```

Reply:
> I'm pretty sure we don't need the `hasattr()` call?

It's a private field, and it could be removed in the future when `Message.interaction` is removed.